alacritty/src/display/mod.rs

```diff
@@ -900,15 +900,17 @@ impl Display {
         };
 
         // Handle IME.
-        if self.ime.is_enabled() {
-            if let Some(point) = ime_position {
+        if let Some(point) = ime_position {
```

Comment:
Winit should indicate that the IME got enabled and you should schedule an update from it, always updating doesn't make any sense and the input into terminal should correctly reposition, since you'll likely have preedit going.

Reply:
I tried a little bit, the `Ime::Enabled` is only generated when the user starts typing, so we have no idea when the user switched the IME. On macOS, when the user is changing the IME (or just toggle caps lock, see the picture below), the text accessory view will appear on the last position you set.

![image](https://github.com/alacritty/alacritty/assets/20237141/3d591c94-3321-47cd-83b8-565a90e13e11)
![Screenshot 2024-01-22 at 17 26 26](https://github.com/alacritty/alacritty/assets/20237141/d36474f8-b778-4fc8-9fc2-09f05867abfe)

So with the current implementation, the user will never get a correct text accessory view position when they are enabling caps lock on macOS.

Reply:
What happens when you press `CapsLock`, do we have any keyboard events in general for it?

Reply:
I mean, it's maybe not delivered from the winit itself. Generally I'm not that much against updating the IME more frequently just on macOS, but I'd like to know a better entry point for such stuff in general.